DataStructures/SortedList/SortedList.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataStructures.SortedList
+{
+    /// <summary>
+    /// Implementation of a Sorted List. It uses Quick Sort Algorithm.
+    /// </summary>
+    /// <typeparam name="T">Generic Type.</typeparam>
+    public class SortedList<T> : IComparisonSorter<T>, IEnumerable<T>
+    {
+        private readonly List<T> list;
+
+        public SortedList() => list = new List<T>();
+
+        public SortedList(List<T> myList)
+        {
+            list = new List<T>(myList.Count);
+            list = myList;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return list.GetEnumerator();
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return ((IEnumerable<T>)list).GetEnumerator();
+        }
+
+        /// <summary>
+        /// Insert item in a sorted list.
+        /// </summary>
+        /// <param name="item">Item of Type T to be inserted in the list.</param>
+        /// <returns>true or false.Item was inserted successfully or not.</returns>
+        public bool Insert(T item)
```

Comment:
Insertion into the sorted list cannot fail because it can contain duplicates

Reply:
done